craft_providers/multipass/multipass_instance.py

```diff
@@ -372,15 +372,16 @@ def pull_file(self, *, source: pathlib.PurePath, destination: pathlib.Path) -> N
     def push_file(self, *, source: pathlib.Path, destination: pathlib.PurePath) -> None:
         """Copy a file from the host into the environment.
 
-        The destination file is overwritten if it exists.
+        The destination file is overwritten if it exists. File permissions are retained
+        but the ownership is changed to the default user `ubuntu`.
 
         :param source: Host file to copy.
         :param destination: Target environment file path to copy to.  Parent
             directory (destination.parent) must exist.
 
         :raises FileNotFoundError: If source file or destination's parent
             directory does not exist.
-        :raises MultipassError: On unexpected error copying file.
+        :raises MultipassError: If the file cannot be pushed into the instance.
```

Comment:
Do we want to provide other appropriate `OSError` subclasses, or should there never be other issues like permission errors?

Reply:
I believe everything else will be caught and raised by Multipass.  For example, if there is a permission error on the source file, Multipass will fail and it will percolate up from `subprocess.CalledProcessError` to a `MultipassError`.

I did add an extra check to confirm the source is not a directory.

Reply:
A part of me feels like I'd like to have more specific errors for what happens in the destination (e.g. an exception class that subclasses both `MultipassError` and `OSError` when the multipass issue is related to OS stuff), but I don't actually know whether that's useful or just me being naggy, so I'm gonna drop it for now :-)